--- a/google-modules/soc/gs/drivers/thermal/samsung/gs_tmu_v3.c
+++ b/google-modules/soc/gs/drivers/thermal/samsung/gs_tmu_v3.c
@@ -313,7 +313,7 @@ static u8 get_therm_press(struct gs_tmu_data *data)
 	return readb(acpm_gov_common.sm_base + thermal_state_offset + therm_press_offset);
 }
 
-int exynos_acme_ect_freq(int cpu, u8 cdev_index);
+int __attribute__((weak)) exynos_acme_ect_freq(int cpu, u8 cdev_index) { return 0; }
 unsigned int gs_tmu_throt_freq(int cpu)
 {
 	struct gs_tmu_data *data = READ_ONCE(per_cpu(pcpu_tmu_data, cpu));
--- a/scripts/link-vmlinux.sh
+++ b/scripts/link-vmlinux.sh
@@ -74,7 +74,9 @@
 		objs="${objs} .vmlinux.export.o"
 	fi
 
-	objs="${objs} init/version-timestamp.o"
+	if [ "$objs" != "vmlinux.o" ]; then
+		objs="${objs} init/version-timestamp.o"
+	fi
 
 	if [ "${SRCARCH}" = "um" ]; then
 		wl=-Wl,
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -77,7 +77,6 @@
 			# If only the short version is requested, don't bother
 			# running further git commands
 			if $short; then
-				echo "+"
 				return
 			fi
 			# If we are past a tagged commit (like
